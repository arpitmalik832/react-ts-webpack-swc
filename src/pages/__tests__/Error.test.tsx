import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import Error from '../Error';

describe('Error component tests', () => {
  afterEach(() => {
    cleanup();
  });

  it('renders without crashing', () => {
    const { container } = render(<Error />);
    expect(container).toBeTruthy();
  });

  it('renders with correct text content', () => {
    const { getByText } = render(<Error />);
    expect(getByText('error')).toBeInTheDocument();
  });

  it('renders a div element', () => {
    const { container } = render(<Error />);
    const divElement = container.querySelector('div');
    expect(divElement).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<Error />);
    expect(container).toMatchSnapshot();
  });

  it('has correct structure', () => {
    const { container } = render(<Error />);
    const divElement = container.firstChild as HTMLElement;

    // Check if it's a div
    expect(divElement.tagName).toBe('DIV');

    // Check text content
    expect(divElement.textContent).toBe('error');

    // Check that it has no child elements (only text)
    expect(divElement.children.length).toBe(0);
  });

  it('is accessible', () => {
    const { container } = render(<Error />);
    // Check if the error message is visible to screen readers
    expect(container.firstChild).toBeVisible();
  });
});
