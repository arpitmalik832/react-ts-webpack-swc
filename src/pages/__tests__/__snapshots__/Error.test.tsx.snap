// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component tests matches snapshot 1`] = `
<div>
  <div>
    error
  </div>
</div>
`;
